### ISSUE TITLE
Should break inside destruction instead of RHS

### ISSUE THREAD

**Issue by so1ve** [prettier/prettier#15407]
**Prettier 3.0.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAExMDMIQA0mARgIYBOmAvpgLx4EDcAOlJFBto0aZceQBeNejyYhCICAAcYAS3TJQlChADuABUoI0yEGQA2asgE9dkkhTJgA1nBgBlMgFs4AGTlQ4yXIbRwLK1t7B2lrTwBzZBgKAFcAkH9nOWi4hLRIgzgARViIeB8-BIArNAAPB0ycvIKkXwN-SQBHGrgNVWldFDI0AFovOAATIYkQGLI5A0iAYQhnZzI9QwNRjKgIrIBBGBi5Elj4DTgKDy9ChoSACxhnAwB1S7l4NDCwOAcdJ7kANyeTPTAaHMIG+8QAklBhrAHGAKHJZJtIQ4YCYsudGiBpKp-HcrNI9Fi4P4KN9vJJPMSYO0yBEFuiEmEKMS9CjpETYfCYKMKHAWnIedTaYs6kVJDAyCQ7nJBjBLsgAEySWL+AAqEq69QxcGcJCGw0GbjI61iNLgADEIBQFjtIksDhBRljPDApTK5Uh5QAGajUIA)

```sh
--parser typescript
--print-width 20
```

**Input:**

```tsx
const { foo, bar } = foo;
const { foo, bar, baz } = foo;
```

**Output:**

```tsx
const { foo, bar } =
  foo;
const {
  foo,
  bar,
  baz,
} = foo;

```

**Expected behavior:**

Should be consistent

#15372

**Comment by g-cappai** [comment]
Hi guys. I'd like to make my first contribution to Prettier. As you can see I already opened a PR, but I need some advice on the expected output, if anyone could take a look. Thank you.

**Comment by seiyab** [comment]
I saw your PR and it seems to improve some format but can also somewhat get worse.
For example,
```diff
- const [width = nextWidth, height = nextHeight, baseline = nextBaseline] =
-   measureText(nextText, getFontString(element));
+ const [
+   width = nextWidth,
+   height = nextHeight,
+   baseline = nextBaseline,
+ ] = measureText(nextText, getFontString(element));
```
looks pretty fine but
```diff
- const {
-   foo,
-   bar: bazAndSomething,
-   quxIsLong,
- } = someBigFunctionName("foo")("bar");
+ const { foo, bar: bazAndSomething, quxIsLong } = someBigFunctionName("foo")(
+   "bar",
+ );
```
looks not good.

I wonder if some conservative change exist and hope related formats to be improved  step-by-step.

**Comment by g-cappai** [comment]
I totally agree. We should define how to handle each case based on the complexity of the lhs. 

Maybe starting from assigning this behavior only to simple variable destructuring and then progeressively decide how to proceed with the other cases may be a good approach.
